src/Microsoft.ML.Transforms/MissingValueDroppingTransformer.cs

```diff
@@ -65,275 +74,330 @@ private static VersionInfo GetVersionInfo()
-        }
+        protected override IRowMapper MakeRowMapper(Schema schema) => new Mapper(this, schema);
 
-        protected override Delegate GetGetterCore(IChannel ch, IRow input, int iinfo, out Action disposer)
+        private sealed class Mapper : MapperBase
         {
-            Host.AssertValueOrNull(ch);
-            Host.AssertValue(input);
-            Host.Assert(0 <= iinfo && iinfo < Infos.Length);
-            Host.Assert(Infos[iinfo].TypeSrc.IsVector);
-
-            disposer = null;
-            Func<IRow, int, ValueGetter<VBuffer<int>>> del = MakeVecGetter<int>;
-            var methodInfo = del.GetMethodInfo().GetGenericMethodDefinition().MakeGenericMethod(Infos[iinfo].TypeSrc.ItemType.RawType);
-            return (Delegate)methodInfo.Invoke(this, new object[] { input, iinfo });
-        }
+            private readonly MissingValueDroppingTransformer _parent;
 
-        private ValueGetter<VBuffer<TDst>> MakeVecGetter<TDst>(IRow input, int iinfo)
-        {
-            var srcGetter = GetSrcGetter<VBuffer<TDst>>(input, iinfo);
-            var buffer = default(VBuffer<TDst>);
-            var isNA = (InPredicate<TDst>)_isNAs[iinfo];
-            var def = default(TDst);
-            if (isNA(in def))
-            {
-                // Case I: NA equals the default value.
-                return
-                    (ref VBuffer<TDst> value) =>
-                    {
-                        srcGetter(ref buffer);
-                        DropNAsAndDefaults(ref buffer, ref value, isNA);
-                    };
-            }
+            private readonly ColumnType[] _srcTypes;
+            private readonly int[] _srcCols;
+            private readonly ColumnType[] _types;
+            private readonly Delegate[] _isNAs;
 
-            // Case II: NA is different form default value.
-            Host.Assert(!isNA(in def));
-            return
-                (ref VBuffer<TDst> value) =>
+            public Mapper(MissingValueDroppingTransformer parent, Schema inputSchema)
+              : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                _parent = parent;
+                _types = new ColumnType[_parent.ColumnPairs.Length];
+                _srcTypes = new ColumnType[_parent.ColumnPairs.Length];
+                _srcCols = new int[_parent.ColumnPairs.Length];
+                _isNAs = new Delegate[_parent.ColumnPairs.Length];
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
                 {
-                    srcGetter(ref buffer);
-                    DropNAs(ref buffer, ref value, isNA);
-                };
-        }
-
-        private void DropNAsAndDefaults<TDst>(ref VBuffer<TDst> src, ref VBuffer<TDst> dst, InPredicate<TDst> isNA)
-        {
-            Host.AssertValue(isNA);
+                    inputSchema.TryGetColumnIndex(_parent.ColumnPairs[i].input, out _srcCols[i]);
```

Comment:
>inputSchema.TryGetColumnIndex(_parent.ColumnPairs[i].input, out _srcCols[i]); [](start = 20, length = 77)

schema validation

Reply:
We need two things: it should be vector, and we should be able to get NaDelegate for it.
Vector check done through CheckInputColumn in MapperBase.
NaDelegate check is done inside GetIsNaDelegate routine.

---
In reply to: [233695475](https://github.com/dotnet/machinelearning/pull/1624#discussion_r233695475) [](ancestors = 233695475)